DevProxy/Proxy/ProxyEngine.cs

```diff
@@ -615,21 +615,44 @@ private static void ToggleSystemProxy(ToggleSystemProxyAction toggle, string? ip
         process.WaitForExit();
     }
 
-    private static bool IsFirstRun()
+    internal static class HasRunFlag

```

Comment:
Let's move this to a separate file for clarity

Reply:
I moved it back to DevProxy since it's not something that we need to expose outside of the runtime for now.